### PR TITLE
[wasm] Jiterpreter cleanup

### PR DESCRIPTION
**Pull request by kg**
* Expand options.h api with ability to dump options as a JSON blob.
* Migrate jiterpreter configuration to options.h
* Disable jiterpreter automatically after multiple JIT failures
Blocks #78428

### CODE REVIEW

**Comment by kg** [comment]
Going to clean up some formatting issues once all the test lanes run, but this is ready for review

File: src/mono/mono/utils/options.c [hunk 1]
```diff
@@ -221,3 +248,94 @@ mono_options_parse_options (const char **argv, int argc, int *out_argc, MonoErro
...
+
+void
+mono_options_get_as_json (char *result_buffer, int result_buffer_size)
+{
+	char *end = result_buffer + result_buffer_size - 1,
+		*destination = result_buffer;
+	gboolean need_comma = FALSE;
+
+	*destination = 0;
+	strncpy (destination, "{\n", result_buffer_size);
```
Comment (lambdageek) [inline src/mono/mono/utils/options.c:325]: what do you think about using `GString` and functions like `g_string_append_printf` to create a temporary result and then copy it into `destination`?

Comment (kg) [reply src/mono/mono/utils/options.c:325]: Sounds good, I dug around in the headers and didn't find a g_ API that looked relevant for this situation but that might be a good one. I'll give it a look if you want me to rework it to use it.

File: src/mono/mono/utils/options.c [hunk 2]
```diff
@@ -249,93 +249,78 @@ mono_options_parse_options (const char **argv, int argc, int *out_argc, MonoErro
...
 					break;
 			}
 
 			src++;
 		}
 
-		destination = strncat_safe (destination, "\"", end);
-		return destination;
+		g_string_append (destination, "\"");
+		return;
```
Comment (vargaz) [inline src/mono/mono/utils/options.c:294]: Might be better to use break here.

File: src/mono/mono/utils/options.c [hunk 3]
```diff
@@ -249,93 +249,78 @@ mono_options_parse_options (const char **argv, int argc, int *out_argc, MonoErro
...
 	}
 
 	default:
 		g_assert_not_reached ();
-		return NULL;
+		return;
```
Comment (vargaz) [inline src/mono/mono/utils/options.c:299]: Ditto.

**Review by vargaz — APPROVED** [review]

**Comment by vargaz** [comment]
The mono changes look ok to me.

**Pull request merged**